login-workflow/src/new-architecture/components/ChangePasswordDialog/ChangePasswordDialog.tsx

```diff
@@ -0,0 +1,103 @@
+        for (let i = 0; i < passwordRequirements.length; i++) {
+            if (!new RegExp(passwordRequirements[i].regex).test(passwordInput)) return false;
+        }
+        return confirmInput === passwordInput;
+    }, [passwordRequirements, passwordInput, confirmInput]);
+
+    const updateFields = useCallback(
+        (fields: { password: string; confirm: string }) => {
+            setPasswordInput(fields.password);
+            setConfirmInput(fields.confirm);
+        },
+        [setPasswordInput, setConfirmInput]
+    );
+
+    const currentPasswordChange = (currentPwd: string): void => {
+        setCurrentInput(currentPwd);
+    };
+
+    useEffect(() => {
+        setPasswordInput(areValidMatchingPasswords() ? passwordInput : '');
+    }, [setPasswordInput, passwordInput, confirmInput, areValidMatchingPasswords]);
+
+    const checkPasswords =
+        passwordInput !== '' && confirmInput !== '' && currentInput !== '' && passwordInput === confirmInput;
+
+    const changePasswordSubmit = (): void => {
+        if (checkPasswords) {
+            try {
+                setIsLoading(true);
+                void actions().changePassword(currentInput, passwordInput);
+                setIsLoading(false);
+            } catch {
+                setShowErrorDialog(true);
+            }
+        }
+    };
+
+    const {
+        dialogTitle = t('bluiAuth:CHANGE_PASSWORD.PASSWORD'),
+        dialogDescription = t('bluiAuth:CHANGE_PASSWORD.PASSWORD_INFO'),
+        currentPasswordLabel = t('bluiCommon:LABELS.CURRENT_PASSWORD'),
+        previousLabel = t('bluiCommon:ACTIONS.BACK'),
+        nextLabel = t('bluiCommon:ACTIONS.OKAY'),
+        onPrevious = (): void => navigate(routeConfig.LOGIN),
+        passwordProps = {
+            newPasswordLabel: t('bluiAuth:CHANGE_PASSWORD.NEW_PASSWORD'),
+            confirmPasswordLabel: t('bluiAuth:CHANGE_PASSWORD.CONFIRM_NEW_PASSWORD'),
+            onPasswordChange: updateFields,
+            passwordRef,
+            confirmRef,
+            initialNewPasswordValue: passwordInput,
+            initialConfirmPasswordValue: confirmInput,
+            passwordRequirements,
+            onSubmit: changePasswordSubmit,
+            passwordNotMatchError: t('bluiCommon:FORMS.PASS_MATCH_ERROR'),
+        },
+        errorDialogProps = {
+            open: showErrorDialog,
+            title: t('bluiCommon:MESSAGES.ERROR'),
+            body: t('bluiAuth:CHANGE_PASSWORD.PROBLEM_OCCURRED'),
+            onClose: (): void => setShowErrorDialog(false),
+            dismissButtonText: t('bluiCommon:ACTIONS.OKAY'),
+        },
+    } = props;
+
+    return (
+        <ChangePasswordDialogBase
+            loading={isLoading}
+            dialogTitle={dialogTitle}
+            dialogDescription={dialogDescription}
+            currentPasswordLabel={currentPasswordLabel}
+            previousLabel={previousLabel}
+            nextLabel={nextLabel}
+            currentPasswordChange={currentPasswordChange}
```

Comment:
`currentPasswordChange` function is only being used once, so no need to define a function, you can directly setState here:
`currentPasswordChange={(currentPwd) => setCurrentInput(currentPwd)}`